src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -969,7 +970,7 @@ address Assembler::locate_operand(address inst, WhichOperand which) {
       assert(which == call32_operand, "jcc has no disp32 or imm");
       return ip;
     default:
-      ShouldNotReachHere();
+      fatal("not handled: 0x0F%2X", 0xFF & *(ip-1));
```

Comment:
I've improved the error reporting a bit here to print out the problematic instruction. I was only able to reproduce the crash through GHA.